Realm/Realm/DatabaseTypes/Accessors/ManagedAccessor.cs

```diff
@@ -181,12 +181,15 @@ public void SubscribeForNotifications(Action<string> notifyPropertyChangedDelega
                 throw new RealmFrozenException("It is not possible to add a change listener to a frozen RealmObjectBase since it never changes.");
             }
 
+            PropertyType collectionFlag = PropertyType.Array | PropertyType.Dictionary | PropertyType.Set;
```

Comment:
Probably we also need to check backlinks? I'm not sure we get notifications for those in any case.

Also, it would make sense to calculate those indices only once, they are not going to change

Reply:
Backlinks are in a separate vector in the Object Store `ObjectSchema` so even if they cause notifications they need to be handled separately. But we need to see if they raise notifications in the first place.

Reply:
Added a test to check this, and they don't in our case